tests/ui/sanitizer/cfi-assoc-ty-lifetime-issue-123053.rs

```diff
@@ -0,0 +1,38 @@
+// Regression test for issue 123053, where associated types with lifetimes caused generation of the
+// trait object type to fail, causing an ICE.
+//
+//@ needs-sanitizer-cfi
+//@ compile-flags: -Ccodegen-units=1 -Clto -Ctarget-feature=-crt-static -Zsanitizer=cfi --edition=2021
+//@ no-prefer-dynamic
+//@ only-x86_64-unknown-linux-gnu
+//@ build-pass
+
+trait Iterable {
```

Comment:
`Iterable` isn't object safe -- no traits with GATs are object safe.

Just use `tcx.object_safety_violations(trait_def_id).is_empty()` in addition to the `is_vtable_safe_method` check, since it actually doesn't check the trait is object safe lol

Reply:
Done.